detectron2/data/dataset_mapper.py

```diff
@@ -138,7 +139,12 @@ def _transform_annotations(self, dataset_dict, transforms, image_shape):
         # bounding box of the cropped triangle should be [(1,0),(2,1)], which is not equal to
         # the intersection of original bounding box and the cropping box.
         if self.recompute_boxes:
+            # FYI: crashes here when filter_empty = False!
+            # instances.gt_boxes = instances.gt_masks.get_bounding_boxes()
+            if not instances.has('gt_masks'): 
+                instances.gt_masks = PolygonMasks([])  # for negative examples
```

Comment:
```suggestion
        if self.recompute_boxes and instances.has('gt_masks'):
```
this should be sufficient

Reply:
I have added negative samples to my training, how do I do the loss function improvement, I run the code and it comes up with Loss became infinite or NaN at iteration=1!

Reply:
One question, in detectron2 I set the background class to id=1,isthing = 1,will this have an effect on the model training, I see that generally the background class is set to id = 0, and also I would like to ask if there is a requirement for the json file format, because generally there is no ANNOTATIONS option for the negative samples, and I added my own bbox = [0,0,0 ,0],and then that seg = [],I just don't know how to improve it now. So that negative samples can be successfully added for training.